test/merge.js

```diff
@@ -0,0 +1,208 @@
+            lon: '-147.6',
+            changeset: '15'
+          },
+          version: versions.B[0]
+        },
+        {
+          type: 'osm/element',
+          id: 'C',
+          links: [versions.C[1], versions.C[2]].sort(),
+          element: {
+            type: 'node',
+            lat: '62.3',
+            lon: '-146.4',
+            changeset: '17'
+          },
+          version: versions.C[3]
+        },
+        {
+          type: 'osm/element',
+          id: 'D',
+          links: [],
+          element: {
+            type: 'way',
+            refs: [ 'A', 'B', 'C' ],
+            changeset: '15'
+          },
+          version: versions.D[0]
+        }
+      ].sort(idcmp)
+      osm0.query(q0, function (err, res) {
+        t.ifError(err)
+        t.deepEqual(sortLinks(res).sort(idcmp), ex0, 'updated query 0 (osm0)')
+        t.deepEqual(res.map(idof).sort(), ex0.map(idof).sort())
+      })
+      osm1.query(q0, function (err, res) {
+        t.ifError(err)
+        t.deepEqual(sortLinks(res).sort(idcmp), ex0, 'updated query 0 (osm1)')
+        t.deepEqual(res.map(idof).sort(), ex0.map(idof).sort())
+      })
+      var q1 = [-149.5, 62, -146, 64]
+      var ex1 = [
+        {
+          type: 'osm/element',
+          id: 'A',
+          links: [],
+          element: {
+            type: 'node',
+            lat: '64.5',
+            lon: '-147.3',
+            changeset: '15'
+          },
+          version: versions.A[0]
+        },
+        {
+          type: 'osm/element',
+          id: 'B',
+          links: [],
+          element: {
+            type: 'node',
+            lat: '63.9',
+            lon: '-147.6',
+            changeset: '15'
+          },
+          version: versions.B[0]
+        },
+        {
+          type: 'osm/element',
+          id: 'C',
+          links: [versions.C[1], versions.C[2]].sort(),
+          element: {
+            type: 'node',
+            lat: '62.3',
+            lon: '-146.4',
+            changeset: '17'
+          },
+          version: versions.C[3]
+        },
+        {
+          type: 'osm/element',
+          id: 'D',
+          links: [],
+          element: {
+            type: 'way',
+            refs: [ 'A', 'B', 'C' ],
+            changeset: '15'
+          },
+          version: versions.D[0]
+        }
+      ].sort(idcmp)
+      osm0.query(q1, function (err, res) {
+        t.ifError(err)
+        t.deepEqual(sortLinks(res).sort(idcmp), ex1, 'updated query 1 (osm0)')
+        t.deepEqual(res.map(idof).sort(), ex1.map(idof).sort())
+      })
+      osm1.query(q1, function (err, res) {
+        t.ifError(err)
+        t.deepEqual(sortLinks(res).sort(idcmp), ex1, 'updated query 1 (osm1)')
+        t.deepEqual(res.map(idof).sort(), ex1.map(idof).sort())
+      })
+    }
+  })
+})
+
+function idcmp (a, b) {
+  if (a.id === b.id) return a.version < b.version ? -1 : 1
+  return a.id < b.id ? -1 : 1
+}
+
+function replicate (osm0, osm1, cb) {
```

Comment:
these'd be handy utilities to put in e.g. `test/utils.js`; they seem to be used by several of the tests.